src/addon-manager-proxy.ts

```diff
@@ -302,7 +321,7 @@ class AddonManagerProxy extends EventEmitter {
     // or don't need a device object.
 
     const adapterId = msg.data.adapterId;
-    const adapter = this.adapters.get(adapterId);
+    const adapter: (MockAdapter & Adapter) | undefined = <MockAdapter & Adapter>this.adapters.get(adapterId);
```

Comment:
Why not simply add lacking methods to `Adapter` class?

Reply:
That's an option but may end up being confusing for add-on developers.

Reply:
Because they are not part of the adapter.
They only exist in the mock.